### PR TITLE
Update SSL/TLS networking tutorial

### PR DESCRIPTION
**Pull request by Faless**
Swap TLS and SSL mentions (TLS is now the most widely accepted term).

Clarify that Godot now uses the trusted certifcates from the operating system by default, and only uses the Mozilla CA bundle as a fallback.

Remove outdated export filter reference (certificates are recognized as resources).

Warn about self-signed certificates still validating domain name when set in project settings (and reference TLSOptions for custom behavior).

Add reference to Crypto.generate_self_signed_certificate and provide further reference about the OpenSSL documentation.

Fixes https://github.com/godotengine/godot/issues/85953

### CODE REVIEW

**Review by mhilbrunner — APPROVED** [review]

File: tutorials/networking/ssl_certificates.rst [hunk 1]
```diff
@@ -73,7 +67,17 @@ Settings.
...
 
-OpenSSL has `some documentation
+.. warning::
+
+    When specifying a self-signed certificate as TLS bundle in the project
+    settings, normal domain name validation is enforced via the certificate
+    :abbr:`CN (common name)` and alternative names. See
+    :ref:`TLSOptions <class_TLSOptions>` to customize domain name validation.
+
+For development purposes Godot can generate self signed certificates via
```
Comment (AThousandShips) [inline tutorials/networking/ssl_certificates.rst:77]:
```suggestion
For development purposes Godot can generate self-signed certificates via
```
Used everywhere else here

File: tutorials/networking/ssl_certificates.rst [hunk 2]
```diff
@@ -1,45 +1,39 @@
...
-   :align: center
-   :alt: Adding ``*.crt`` to non-resource export filter in the export preset
-
-   Adding ``*.crt`` to non-resource export filter in the export preset
-
 There are two ways to obtain certificates:
 
-Acquire a certificate from a certificate authority
+Obtain a certificate from a certificate authority
 --------------------------------------------------
```
Comment (AThousandShips) [inline tutorials/networking/ssl_certificates.rst:37]:
```suggestion
-------------------------------------------------
```

**Review by AThousandShips — APPROVED** [review]

**Review by mhilbrunner — APPROVED** [review]

**Comment by mhilbrunner** [comment]
Thank you, merged.

**Comment by NervenCid** [comment]
> Swap TLS and SSL mentions (TLS is now the most widely accepted term).
> 
> Clarify that Godot now uses the trusted certifcates from the operating system by default, and only uses the Mozilla CA bundle as a fallback.
> 
> Remove outdated export filter reference (certificates are recognized as resources).
> 
> Warn about self-signed certificates still validating domain name when set in project settings (and reference TLSOptions for custom behavior).
> 
> Add reference to Crypto.generate_self_signed_certificate and provide further reference about the OpenSSL documentation.
> 
> Fixes [godotengine/godot#85953](https://github.com/godotengine/godot/issues/85953)

Hello, i have a question: 

That works for websockets in WSS too?

**Comment by mhilbrunner** [comment]
Cherry-picked to 4.3 in https://github.com/godotengine/godot-docs/pull/10038.

**Pull request merged**